velox/type/parser/TypeParser.yy

```diff
@@ -104,8 +104,8 @@ array_type : ARRAY LPAREN type RPAREN             { $$ = ARRAY($3); }
            | ARRAY LPAREN type_with_spaces RPAREN { $$ = ARRAY(inferTypeWithSpaces($3, true).second); }
            ;
 
-map_type : MAP LPAREN single_word_type COMMA type RPAREN             { $$ = MAP($3, $5); }
-         | MAP LPAREN single_word_type COMMA type_with_spaces RPAREN { $$ = MAP($3, inferTypeWithSpaces($5, true).second); }
+map_type : MAP LPAREN type COMMA type RPAREN                         { $$ = MAP($3, $5); }
```

Comment:
I thought the map key was always a simple/scalar type.
We don't need the `single_word_type` nonterminal if this is the case. Can you remove `single_word_type` and inline its rules in the `type` nonterminal?

Reply:
Map keys can be complex type:

```
presto:tpch> select cast(null as map(map(integer, varchar), varchar));
 _col0
-------
 NULL
(1 row)
```